tests/nc-client/main.c

```diff
@@ -16,11 +18,40 @@ struct senddata {
 	int conndone;
 	void * connect_cookie;
 	void * write_cookie;
+	void * read_cookie;
+	uint8_t shutdown;
 };
 
 /* Forward declaration. */
 static int callback_wrote(void *, ssize_t);
 
+static int callback_stopping(void * cookie, ssize_t lenread)
+{
+	struct senddata * send = (struct senddata *)cookie;
+
+	/* We are no longer reading. */
+	send->read_cookie = NULL;
+
+	/* Check results. */
+	if (lenread == -1) {
+		warnp("network_read received");
+		goto err0;
+	} else {
+		/* We should have received nothing. */
+		assert(lenread == 0);
```

Comment:
This shouldn't be an assertion: We're asking network_read to read a byte of data, so if it returns 1 it doesn't indicate that there is a bug *internal to this program*.  (However, it indicates that the server is misbehaving, since the server shouldn't send any data back; so we should treat a nonzero value as an error.)